debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/MethodForNameCommand.java

```diff
@@ -68,85 +68,74 @@ public void run(String command, String[] args, Context context, PrintStream out)
 	}
 
 	int dbgGetMethodsForName(PrintStream out, String pattern) throws CorruptDataException {
-		int matchCount = 0;
-		// J9ClassWalkState walkState;
-		String classNeedle, methodNeedle, sigNeedle;
-		long classMatchFlags, methodMatchFlags, sigMatchFlags;
-		String classStart, nameStart, sigStart;
-
-		if (pattern.indexOf('.') != -1) {
-			nameStart = pattern.substring(pattern.indexOf('.') + 1); // skip the .
-			classStart = pattern.substring(0, pattern.indexOf('.'));
-		} else {
-			classStart = "*";
-			nameStart = pattern;
-		}
+		int dot = pattern.indexOf('.');
+		String classStart = (dot == -1) ? "*" : pattern.substring(0, dot);
+		int paren = pattern.indexOf('(', dot + 1);
+		String nameStart = (paren == -1) ? pattern.substring(dot + 1) : pattern.substring(dot + 1, paren);
+		String sigStart = (paren == -1) ? "*" : pattern.substring(paren);
+		StringBuffer needleBuffer;
 
-		if (pattern.indexOf('(') != -1) {
-			sigStart = pattern.substring(pattern.indexOf('('));
-		} else {
-			sigStart = "*";
-		}
-
-		StringBuffer needleBuffer = new StringBuffer();
-		classMatchFlags = WildCard.parseWildcard(classStart, needleBuffer);
-		classNeedle = needleBuffer.toString();
+		needleBuffer = new StringBuffer();
```

Comment:
Just curious why you split the declaration and the initialization.

Reply:
`needleBuffer` is reused several times: in my opinion, it's misleading to put the first assignment into the declaration.